### PR TITLE
Hide Events Collection behind feature flag 

### PR DESCRIPTION
**Pull request by Harriethw**
## Description of change

We are going to be user testing a feature where we show Events via Activity Stream, rather than the Data Hub API. This is so we can return Aventri Event objects as well as Data Hub events. 
The first step in this is checking whether the user feature flag is on, and turning off the old collection list if it is. 

## Test instructions

- add the `user-event-activities` flag to your adviser in the django admin
- visit the Events tab and see that it's a blank page
- when the flag is off everything should be as normal

### Before
Screenshot 2022-06-10 at 15 42 10

### After

Screenshot 2022-06-10 at 15 44 55

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"

- [ ] Has the branch been rebased to master?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

### CODE REVIEW

File: src/apps/contacts/router.js [hunk 1]
```diff
@@ -43,21 +46,21 @@
...
   '/:contactId/audit',
-  userFeatures('user-contact-activities'),
+  userFeatures(CONTACT_ACTIVITY_FEATURE_FLAG),
   getAudit
 )
 
 router.get(
   '/:contactId/documents',
-  userFeatures('user-contact-activities'),
+  userFeatures(CONTACT_ACTIVITY_FEATURE_FLAG),
```
[the remark this answers is not in the document]
Comment (Harriethw) [reply src/apps/contacts/router.js:63]: this was obviously flagged before, so I think...we just ignore it?

[the remark this answers is not in the document]
Comment (cgsunkel) [reply src/apps/contacts/router.js:63]: Yes we can ignore all these alerts, I think TechEx were going to investigate but they haven't got around to it yet

**Review by craigjbass — APPROVED** [review]

File: src/client/modules/Events/CollectionList/index.jsx [hunk 2]
```diff
@@ -77,93 +78,99 @@ const EventsCollection = ({
...
-              placeholder="Search UK region"
-              options={optionMetadata.ukRegionOptions}
-              selectedOptions={selectedFilters.ukRegions.options}
-              data-test="uk-region-filter"
-            />
-          </FilterToggleSection>
-        </CollectionFilters>
-      </FilteredCollectionList>
+      <CheckUserFeatureFlag userFeatureFlagName={EVENT_ACTIVITY_FEATURE_FLAG}>
+        {(isfeatureflagOn) =>
```
Comment (tomkingston) [inline src/client/modules/Events/CollectionList/index.jsx:82]: Tiny thing, but you might want to have this whole variable in camelCase - so `isFeatureFlagOn`

**Review by richmadetech — APPROVED** [review]

**Review by tomkingston — APPROVED** [review]
Looks good - but I think it's possible you might end up with the page flashing when it makes the whoami call to get the feature flags. You might need to get the feature flags on the express layer to avoid any flashing.

**Comment by Harriethw** [comment]
@tomkingston thanks for flagging - we wanted to find a solution that wouldn't require express as the Events collection doesn't go through the express middleware atm. I think thanks to the `Resource` component and React rendering cycle we've mitigated any flashing, and it shows the loading spinner before rendering, but we'll keep an eye on it 👍

**Pull request merged**